### ISSUE TITLE
Loading and detecting jQuery in multiversion context IE8 bug

### ISSUE THREAD

**Issue by madrussa** [requirejs/requirejs#1536]
Hi, I've been working on trying to detect and only load jQuery when it hasn't been loaded in the page already. I used the method given before in other tickets:
```
if (typeof window.jQuery === 'function') { 
    define('jquery', function() { return window.jQuery; }); 
}
```

However because I'm using a custom context I had to do a little more to append jQuery to my configuration:
```
var myConfig = {
    context: "myRequire",
    baseUrl: '/js',
    paths: {/*...*/}
};
if (typeof window.jQuery === 'function') { 
    define('jquery', function() { return window.jQuery; }); 
} else {
    myConfig.paths.jquery = '//ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min';
}
var myRequire = require.config(myConfig);
```

This happens to work fine in Chrome, Firefox and such but in IE8 and below it wasn't including jquery in my application.
```
myRequire(['require', 'jquery'], function(require, jquery) {
    console.log(jquery); // undefined in IE8
});
```

Is there a way of defining modules for specific contexts? I have a work around but because the behaviour seems to differ between browsers I thought I should still raise an issue.

My work around for anyone else with a similar issue was to create a jQuery loader file which because it's loaded within the same context seems to work ok across all browsers:

**main.js**
```
var myConfig = {
    context: "myRequire",
    baseUrl: '/js',
    paths: {/*...*/}
};
if (typeof window.jQuery === 'function') { 
    myConfig.map = {
        '*' : {
            'jquery' : 'jquery-loader'
        }
    }
} else {
    myConfig.paths.jquery = '//ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min';
}
var myRequire = require.config(myConfig);
````
**jquery-loader.js**
```
define(function(require) {
    return window.jQuery;
});
```

Perhaps I've missed something or haven't configured it properly? Any suggestions would be super. Oh and I'm running require version 2.2.0

**Comment by jrburke** [comment]
You have hit a wrinkle with the multiversion support, this is not easy to do today. One thought has been to provide a .define() method on the context-specific require returned from a requirejs.config() call, but no work in that actual direction, and it is unlikely at this point.

Does this work out instead:

```javascript
var myConfig = {
    context: "myRequire",
    baseUrl: '/js',
    paths: {/*...*/
      // Add the path to jQuery now, does not trigger loading.
      jquery: '//ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min'
    }
};
var myRequire = require.config(myConfig);

if (typeof window.jQuery === 'function') { 
    define('jquery', function() { return window.jQuery; }); 
}
myRequire(['jquery']);
```

A `myRequire([])` may also be enough. Basically the idea is to call config to get the context set up, then call define() to get it into the global queue, then call the context-specific myRequire() so that it claims all the define()'s that are sitting in the global queue.

Closing as a discussion ticket, but feel free to continue discussion here.

**Comment by madrussa** [comment]
It does seem odd it works differently in Chrome vs IE8... perhaps it's a scope issue.

Anyway, the method above does work but it's not something we can do as we append jQuery for our components but the client could have their own jQuery on the page. Providing it's using a defined version range we'll use their jQuery if it is included, else we'll include our own. The problem arises when the versions are different or the urls are different, it'll include the jQuery we have defined in the config. Any of our plugins will be bound to our version of jQuery and then the client can't interface with our plugins due to them using their own version... 

The examples I provided were simplified from our actual tests but still presented the same issue. Thanks for taking the time to respond, hopefully the workaround I presented will be useful to other people in a similar situation as the overhead is minor and it works across all the browsers we've tested.
